rewrite-maven/src/main/java/org/openrewrite/maven/search/ModuleHasNoParent.java

```diff
@@ -0,0 +1,97 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven.search;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.ScanningRecipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.marker.JavaProject;
+import org.openrewrite.marker.SearchResult;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ModuleHasNoParent extends ScanningRecipe<ModuleHasNoParent.Accumulator> {
+
+    @Override
+    public String getDisplayName() {
+        return "Module has no parent";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Searches for Maven modules that have no parent. " +
+                "Places a `SearchResult` marker on all sources within a module that has no parent. " +
+                "This recipe is intended to be used as a precondition for other recipes. " +
+                "For example: " +
+                "this could be used to limit the application of recipes that add a parent to projects that currently don't have one.";
+    }
+
+    @Value
+    public static class Accumulator {
+        Set<JavaProject> projectsWithParent;
+    }
+
+    @Override
+    public Accumulator getInitialValue(ExecutionContext ctx) {
+        return new Accumulator(new HashSet<>());
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(Accumulator acc) {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+                assert tree != null;
+                tree.getMarkers()
+                        .findFirst(JavaProject.class)
+                        .ifPresent(jp -> {
+                            Tree t = new ParentPomInsight("*", "*", null, false).getVisitor().visit(tree, ctx);
+                            if (t != tree) {
+                                acc.getProjectsWithParent().add(jp);
+                            }
+                        });
+                return tree;
+            }
+        };
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(Accumulator acc) {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+                assert tree != null;
+                Optional<JavaProject> maybeJp = tree.getMarkers().findFirst(JavaProject.class);
+                if (!maybeJp.isPresent()) {
+                    return tree;
+                }
+                JavaProject jp = maybeJp.get();
+                if (!acc.getProjectsWithParent().contains(jp)) {
```

Comment:
What is the reasoning for this recipe being a scanning recipe?
I don't see why you couldn't just call `parentPomInsight` here and save the scanning cycle of the recipe :)